### ISSUE TITLE
[Bug]: Does not remember setting cron from "Webcron" to "Cron"

### ISSUE THREAD

**Issue by martin21** [nextcloud/server#32610]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

In two Nextcloud 24.0.1 instances I cannot change cron settings from "webcron" to "cron" from within the admin web interface.

# Actual result

I click on "Cron (Recommended)". When I access the page the selected setting still is "Webcron". Additionally I remember having set "Cron (Recommended)" in an earlier version of Nextcloud, in one instance it was Nextcloud 23, in another I don't know the version anymore and it worked. So it appeared to have lost the setting I once successfully made.

### Steps to reproduce

1. On Nextcloud 24.0.1 change cron setting from "Webcron" to "Cron (Recommended)

### Expected behavior

Nextcloud remembers the setting.

### Installation method

Web installer on a VPS or web space

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 7.4

### Web server

Apache (supported)

### Database engine version

MariaDB

### Is this bug present after an update or on a fresh install?

Updated to a major version (ex. 22.2.3 to 23.0.1)

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
I removed domain names cause I see no need to link my github account to some domain names.

Instance 1:

{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "[… also sensitive value …]"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "24.0.1.1",
        "overwrite.cli.url": "[… also sensitive value …]",
        "htaccess.RewriteBase": "\/",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 6379,
            "password": "***REMOVED SENSITIVE VALUE***"
        },
        "installed": true,
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpmode": "smtp",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_sendmailmode": "smtp",
        "default_phone_region": "[…]",
        "maintenance": false,
        "theme": "",
        "logtype": "file",
        "loglevel": 2,
        "logfile": "\/var\/log\/nextcloud.log",
        "updater.release.channel": "stable",
        "app_install_overwrite": [
            "audioplayer"
        ],
        "mail_smtpport": "25",
        "updater.secret": "***REMOVED SENSITIVE VALUE***"
    }
}

Instance 2:

{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "[… also sensitive value …]",
            "[… also sensitive value …]",
            "[… also sensitive value …]"
        ],
        "overwrite.cli.url": "[… also sensitive value …]",
        "htaccess.RewriteBase": "\/",
        "default_phone_region": "[…]",
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "24.0.1.1",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 0
        },
        "installed": true,
        "maintenance": false,
        "theme": "",
        "loglevel": 2,
        "mail_smtpmode": "smtp",
        "mail_sendmailmode": "pipe",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "25",
        "updater.release.channel": "stable",
        "updater.secret": "***REMOVED SENSITIVE VALUE***"
    }
}
```

### List of activated Apps

```shell
No relevance.
```

### Nextcloud Signing status

```shell
That is not working on even when I remove index.php as I use URL rewriting. I just get redirected to the start page of Nextcloud.
```

### Nextcloud Logs

```shell
No log message related to this error in nextcloud.log. In one instances just some random instances of "Trusted domain error" with random source IPs trying to access the Nextcloud instance by its IP address which is of course not listed as trusted domain. In the other instance there is nothing at all in the log. I had logging app disabled there but enabled it again and then went to reproduce the issue. Nothing in log.

Additionally I do not see anything in Apache error logs.
```

**Comment by szaimen** [comment]
Hi, please update to 24.0.9 or better 25.0.3 and report back if it fixes the issue. Thank you!

My goal is to add a label like e.g. 25-feedback to this ticket of an up-to-date major Nextcloud version where the bug could be reproduced. However this is not going to work without your help. So thanks for all your effort!

If you don't manage to reproduce the issue in time and the issue gets closed but you can reproduce the issue afterwards, feel free to create a new bug report with up-to-date information by following this link: https://github.com/nextcloud/server/issues/new?assignees=&labels=bug%2C0.+Needs+triage&template=BUG_REPORT.yml&title=%5BBug%5D%3A+

**Comment by martin21** [comment]
Hi Simon.

Simon L. - 23.01.23, 15:18:19 CET:
> Hi, please update to 24.0.9 or better 25.0.3 and report back if it
> fixes the issue. Thank you!

Appears to be fixed in 25.0.3.

Thanks,
-- 
Martin

**Comment by szaimen** [comment]
Thanks for verifying!
